### DISCUSSION TITLE
Converting Query type from Document to com.mongodb.BasicDBObject

### DISCUSSION THREAD

**Discussion by ritik-jaiswal-miq** [MorphiaOrg/morphia#2835] [Q&A]
Hey @evanchooly  , I am migrating from morphia 1.3.2 to 2.3.9 , I am trying to create query the query seems to be type of Document ,
earlier it used to be com.mongodb.BasicDBObject , how can I make it to be so 
<img width="968" alt="Screenshot 2024-02-02 at 3 44 20 PM" src="https://github.com/MorphiaOrg/morphia/assets/157503200/5d7921e1-d321-4991-90c6-01b20ccaa692">

**Comment by evanchooly** [comment]
`BasicDBObject` is a driver type that has gone away (effectively). The internal representation of the `Query` should never matter to your code. What problem are you seeing?

**Reply by ritik-jaiswal-miq** [reply]
When converted to string Document{{item_type=struct, item_structure=[Document{{name=segment_id, type=bigint}}, Document{{name=segment_modifier, type=bigint}}]}} , when it is passed to object mapper of class of jackson_databind then its throwing errror (com.fasterxml.jackson.databind.ObjectMapper) seems its looking for BasicDBObject instead of document ,  (ObjectNode@7835){"item_type":"struct","item_structure":[{"name":"segment_id","type":"bigint"},{"name":"segment_modifier","type":"bigint"}]}  basicDBObject is coming in earlier branch  which was on morphia 1.3.2
